### PR TITLE
VR-4742: Raise more informative error on non-JSON response bodies

### PR DESCRIPTION
**Pull request by convoliution**
Most of this PR is a find-and-replace of `response.json()` with `_utils.body_to_json(response)` except for my two comments.

### CODE REVIEW

File: client/verta/tests/test_utils.py [hunk 1]
```diff
@@ -13,6 +13,43 @@
 from . import utils
 
 
+class TestBodyToJson:
```
Comment (convoliution) [inline client/verta/tests/test_utils.py:16]: These tests are new

File: client/verta/verta/_internal_utils/_utils.py [hunk 2]
```diff
@@ -306,6 +306,38 @@ def raise_for_http_error(response):
...
+    try:
+        return response.json()
+    except ValueError:  # not JSON response
+        e = ValueError('\n'.join([
+            "expected JSON response from {}, but instead got:".format(response.url),
+            response.text or "<empty response>",
+            "",
+            "Please notify the Verta development team.",
+        ]))
+        six.raise_from(e, None)
```
Comment (convoliution) [inline client/verta/verta/_internal_utils/_utils.py:338]: This is the new util

File: client/verta/verta/_internal_utils/_utils.py [hunk 3]
```diff
@@ -306,6 +306,38 @@ def raise_for_http_error(response):
...
+
+
```
Comment (convoliution) [inline client/verta/verta/_internal_utils/_utils.py:340]: This is the new util

File: client/verta/verta/_internal_utils/_utils.py [hunk 4]
```diff
@@ -306,6 +306,39 @@ def raise_for_http_error(response):
...
+        msg = '\n'.join([
+            "expected JSON response from {}, but instead got:".format(response.url),
+            response.text or "<empty response>",
+            "",
+            "Please notify the Verta development team.",
+        ])
+        msg = six.ensure_str(msg)
+        six.raise_from(ValueError(msg), None)
+
+
```
Comment (convoliution) [inline client/verta/verta/_internal_utils/_utils.py:341]: This is the new util

**Review by conradoverta — APPROVED** [review]
LGTM

**Pull request merged**